### PR TITLE
TR-1347 - Insert snippets in Templates V2

### PR DESCRIPTION
**Pull request by nicklemmon**
[TR-1347](https://jira.int.messagesystems.com/browse/TR-1347)

### What Changed
- Adds popover to the templates editor with "Insert Snippet" action
- Adds new `<InsertSnippetModal/>` component
- Adds supporting tests for both of these changes
- Simplifies `EditSection.module.scss` styles

### How To Test
- Go to `/templatesV2` and click on a template
- Verify that the kebab menu opens the popover
- Verify that the "Insert Snippet" action opens a modal and closes the popover
- Verify that snippets are searchable in the typeahead
- Verify that the selected snippet is populated in the copy field

### To Do
- [ ] Incorporate feedback

### CODE REVIEW

**Comment by nicklemmon** [comment]
> Just a question, the state of 'Add a snippet' modal persists after we select a snippet and close the modal and open it again. Do we want to persist old state or should it change to `{{ render_snippet( "example-id" ) }}`

Yes! It should reset. I'm guessing its not doing that at the moment?

**Comment by nicklemmon** [comment]
Alright, fixed it! Good catch.

File: src/pages/templatesV2/components/EditSection.js [hunk 1]
```diff
@@ -1,24 +1,77 @@
...
     <Panel className={styles.EditSection}>
       <div className={styles.TabsWrapper}>
+        {/* eslint-disable no-unused-vars */}
         <Tabs
           color="blue"
           selected={currentTabIndex}
           onSelect={(nextTabIndex) => { setTab(nextTabIndex); }}
           tabs={tabs.map(({ render, ...tab }) => tab)}
         />
+        {/* eslint-enable no-unused-vars */}
```
Comment (shrirupa) [inline src/pages/templatesV2/components/EditSection.js]: do we need this twice ? (line 30 & 37)

Comment (nicklemmon) [reply src/pages/templatesV2/components/EditSection.js]: I'm disabling an ESLint rule here and then turning it back on again - so yeah! Beginning and end.

**Pull request merged**